app/controllers/sessions_controller.rb

```diff
@@ -100,4 +102,97 @@ def passes_invite_reqs
     invitation = check_user_invited("", session[:invite_token], @user_domain)
     invitation[:present]
   end
+
+  def process_external_signin
+    @user_exists = check_user_exists
+
+    # If using invitation registration method, make sure user is invited
+    return redirect_to root_path, flash: { alert: I18n.t("registration.invite.no_invite") } unless passes_invite_reqs
+
+    @auth['info']['customer'] = parse_user_domain(request.host) if Rails.configuration.loadbalanced_configuration
+    user = User.from_external_provider(@auth)
+
+    # Add pending role if approval method and is a new user
+    if approval_registration && !@user_exists
+      user.add_role :pending
+
+      # Inform admins that a user signed up if emails are turned on
+      send_approval_user_signup_email(user) if Rails.configuration.enable_email_verification
+
+      return redirect_to root_path, flash: { success: I18n.t("registration.approval.signup") }
+    end
+
+    send_invite_user_signup_email(user) if Rails.configuration.enable_email_verification &&
+                                           invite_registration && !@user_exists
+
+    login(user)
+  end
+
+  def send_ldap_request
+    host = ''
+    port = 389
+    bind_dn = ''
```

Comment:
I'm not quite sure that this is the right place to put this code. Maybe move it into a concern? or some sort of module?